### PR TITLE
Feaure/29 add scene switching

### LINKED ISSUE #29: Background-Scene Switching
## Description

As the game has multiple background scenes (bg-scenes), we want to be able to switch between them. When the SCENE command is triggered from the dialogue script, the current bg-scene should be disabled, and the new bg-scene should be enabled.

## Dev Guide

- A bg-scene should consists of three parts: Background, actor, foreground. 
- The background is a still image of the background, and is on render layer 0
- The Actor is an object that the scene manager can always reference to and expose to other scripts which contains a sprite renderer and an animator. It is on render layer 1.
- The foreground is a still or animated image that appears in front of the actor, and is on render layer 2.
- BG-Scenes should be contained within a single gameobject that we can easily turn into a prefab.
- Inactive BG-scenes should keep their assigned actors in there, so we don't have to keep re-initializing those.

## Acceptance Criteria

This story will be finished when we have a sample unity-scene (known as: scene) which shows at least two separate bg-scenes, and can switch between them using an inky script

### PR DESCRIPTION
**Pull request by JuiceGrape**
## Summary
Scenes can now be switched based on a BGSceneList in the scene. Has to be linked to the SceneController. 
It isn't added to standardInclusions prefab, because it would end up very large when all BG-Scenes are in there. Instead, bg-scenes are prefabs itself and should be added under the BGSceneList in the scene.

## Before/after screenshots and/or animated gif
longone

## Testing instructions
Open inky scene and run it.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource: 
  

closes #29

### CODE REVIEW

File: Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaMouse.cs [hunk 1]
```diff
@@ -1,132 +1,132 @@
...
-            RectTransform openSecondSubMenuButton = subMenu.gameObject.GetComponentsInChildren<RectTransform>().First(menuItem => menuItem.gameObject.name == "LoadButton (1)");
-            RectTransform closeSecondSubMenuButton = secondSubMenu.gameObject.GetComponentsInChildren<RectTransform>().First(menuItem => menuItem.gameObject.name == "LoadButton (4)");
-            RectTransform closeFirstSubMenuButton = subMenu.gameObject.GetComponentsInChildren<RectTransform>().First(menuItem => menuItem.gameObject.name == "LoadButton (4)");
-
-            Assert.True(mainMenu.Active);
-            Assert.False(subMenu.Active);
-
-            Set(mouse.position, openFirstSubMenuButton.position + openFirstSubMenuButton.localScale * 0.5f);
-            yield return PressForFrame(mouse.leftButton);
+﻿//using System.Collections;
```
Comment (lyndsiWilliams) [inline Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaMouse.cs]: What happened with this test?

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
Can this test be fixed instead of commented out?

File: Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaMouse.cs (hunk 1 above)
Comment (JuiceGrape) [reply Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaMouse.cs]: Accidentally comitted the commented out one, will revert.

Comment (lyndsiWilliams) [reply Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaMouse.cs]: Oh whoops lol! Looks great otherwise 😁

**Review by lyndsiWilliams — DISMISSED** [review]

File: Assets/Scripts/BGScenes/BGSceneList.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class BGSceneList : MonoBehaviour
+{
+
+    [ReadOnly] private Dictionary<string, BGScene> _scenesInChildren;
```
Comment (PhoebeMitchell) [inline Assets/Scripts/BGScenes/BGSceneList.cs]: What is the readonly attribute for? From what I understand it makes it so you cant change fields in the inspector, but dictionaries don't appear in the inspector.

File: Assets/Scripts/BGScenes/BGSceneList.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    private BGScene _activeScene;
+
+    /// <summary>
+    /// Initalizes the background scene dictionary
```
Comment (PhoebeMitchell) [inline Assets/Scripts/BGScenes/BGSceneList.cs:14]: Spelling: Initializes

File: Assets/Scripts/BGScenes/BGSceneList.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            _scenesInChildren.Add(scene.gameObject.name, scene);
+        }
+    }
+
+    /// <summary>
+    /// Sets the new active scene based on the provided scene name.
+    /// </summary>
+    /// <param name="sceneName">name of the target scene</param>
+    /// <returns>The new active scene. Can be null if an error ocurred.</returns>
```
Comment (PhoebeMitchell) [inline Assets/Scripts/BGScenes/BGSceneList.cs:29]: Spelling: Occurred

File: Assets/Scripts/BGScenes/SceneController.cs [hunk 5]
```diff
@@ -5,6 +5,9 @@
 
 public class SceneController : MonoBehaviour, ISceneController
 {
+    [SerializeField] private BGSceneList _sceneList;
```
Comment (PhoebeMitchell) [inline Assets/Scripts/BGScenes/SceneController.cs:9]: Add tooltip

File: Assets/Scripts/BGScenes/BGSceneList.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public BGScene SetScene(string sceneName) //TODO: Change this when making file names universal
```
Comment (PhoebeMitchell) [inline Assets/Scripts/BGScenes/BGSceneList.cs:30]: Be more specific about what needs to be changed.

**Review by PhoebeMitchell — DISMISSED** [review]

File: Assets/Scripts/BGScenes/BGSceneList.cs (hunk 2 above)
Comment (JuiceGrape) [reply Assets/Scripts/BGScenes/BGSceneList.cs]: Ah yeah, my attempt to debug it, didn't remove it cause I forgot after

File: Assets/Scripts/BGScenes/BGSceneList.cs (hunk 6 above)
Comment (JuiceGrape) [reply Assets/Scripts/BGScenes/BGSceneList.cs:30]: Story details what needs to be changed

**Review by PhoebeMitchell — APPROVED** [review]

**Review by lyndsiWilliams — APPROVED** [review]
Very naise, friend ✨

**Pull request merged**